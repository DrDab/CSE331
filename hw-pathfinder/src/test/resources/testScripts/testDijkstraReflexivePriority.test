# Test if Dijkstra prioritizes the reflexive edge over a cyclic path for a given node.

# Create graph g
CreateGraph g

# Add city names to graph g
AddNode g LosSantos
AddNode g SanAndreas
AddNode g SanFierro
AddNode g LasVenturas
AddNode g Ludendorff
AddNode g Vice
AddNode g Liberty
AddNode g Alderney

# Add edges to graph g
AddEdge g LosSantos SanAndreas 1.23
AddEdge g SanAndreas LosSantos 1.23

# Test pathfinding
FindPath g LosSantos LosSantos
