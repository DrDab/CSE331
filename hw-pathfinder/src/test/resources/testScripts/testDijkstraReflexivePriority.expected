# Test if Dijkstra prioritizes the reflexive edge over a cyclic path for a given node.

# Create graph g
created graph g

# Add city names to graph g
added node LosSantos to g
added node SanAndreas to g
added node SanFierro to g
added node LasVenturas to g
added node Ludendorff to g
added node Vice to g
added node Liberty to g
added node Alderney to g

# Add edges to graph g
added edge 1.230 from LosSantos to SanAndreas in g
added edge 1.230 from SanAndreas to LosSantos in g

# Test pathfinding
path from LosSantos to LosSantos:
total cost: 0.000